bootstrap/bootstrap.sh

```diff
@@ -28,8 +29,54 @@ sudo apt-get update
 sudo apt-get -y dist-upgrade
 
 #----nodejs install
-curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
+curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
 sudo apt-get install -y nodejs
 
-#----alsa install
-sudo apt-get install -y alsa-base alsa-utils
+#----install official requirements
+sudo apt-get install -y alsa-base alsa-utils libasound2-dev
+
+#----install missing pigpio in Raspbian Lite (the command npm install pigpio will be exec by package.json)
+sudo apt-get install pigpio
+
+#----install git and download tjbot
+if [ ! -d "${TJBOT_FOLDER}/recipes/conversation" ]; then
+    sudo apt-get install -y git
+
+    rm -Rf /home/pi/git/tjbot
+    git clone https://github.com/ibmtjbot/tjbot.git /home/pi/git/tjbot
+
+    TJBOT_FOLDER='/home/pi/git/tjbot'
+fi
+
+#----install conversation (to install it will resolve tjbotlib and other dependencies)
+cd $TJBOT_FOLDER
+cd recipes/conversation
+echo "path: $PWD"
+
+sudo npm install --unsafe-perm -g node-gyp
```

Comment:
@bgbraga 

> ```
> gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/6.10.2"
> gyp WARN EACCES attempting to reinstall using temporary dev dir "/home/pi/tjbot/recipes/conversation/node_modules/fibers/.node-gyp"
> ```
The EACCES warnings are occurring because bootstrap.sh was called with sudo from the command line. Try with `./bootstrap.sh ` rather than `sudo ./bootstrap.sh` and the EACCES warnings should no longer occur.

>As we can see in our logs, if we use only npm install we receive a lot of warnings in installation process. I don't know if it is related with a wrong package version or other thing.
>Do you know how to solve these warning messages? I know it is only warning but is a lot of then and give the impression of an installation problem.

I can't tell you if it's related to the wrong package versions. I assume that the authors tjbot and tjbotlib checked that the versions of the packages being used can be used. It's most likely the case that the authors of the native packages simply haven't updated their packages to prevent the warnings in Node.js v6 yet. I think you have the following options:
 * Live with the warnings and document them well so that users feel more at ease
 * Downgrade to Node.js v4 to see if there are no warnings or less warnings
 * Create issues at the corresponding repositories asking the authors if the packages can be updated and maybe create pull requests if they can be updated
 * Hide the output from the user with command line redirection, for example, something like `npm install > install.log 2>&1` might work.